.github/workflows/tests.yml

```diff
@@ -0,0 +1,11 @@
+on: [push, pull_request]
+jobs:
+  test-project:
+    runs-on: ['ubuntu-latest']
```

Comment:
What about macos?

Reply:
Github seems to have problems finding macos runners. Also macos has some strange bug when program halts for no reason in multithreaded/generator mode. I'd say that fixing those problems is not in priority at that moment.

Reply:
probably they limit number of created threads. You should add this as a command line parameter and test.